src/openvic-simulation/map/State.cpp

```diff
@@ -1,23 +1,45 @@
 #include "State.hpp"
 
+#include "openvic-simulation/country/CountryInstance.hpp"
 #include "openvic-simulation/map/MapDefinition.hpp"
 #include "openvic-simulation/map/MapInstance.hpp"
 #include "openvic-simulation/map/ProvinceInstance.hpp"
 #include "openvic-simulation/map/Region.hpp"
+#include "openvic-simulation/utility/StringUtils.hpp"
 
 using namespace OpenVic;
 
 State::State(
-	StateSet const& new_state_set, CountryDefinition const* owner, ProvinceInstance* capital,
-	std::vector<ProvinceInstance*>&& provinces, ProvinceInstance::colony_status_t colony_status
+	StateSet const& new_state_set, CountryInstance* owner, ProvinceInstance* capital,
+	std::vector<ProvinceInstance*>&& provinces, ProvinceInstance::colony_status_t colony_status,
+	decltype(pop_type_distribution)::keys_t const& pop_type_keys
 ) : state_set { new_state_set }, owner { owner }, capital { capital }, provinces { std::move(provinces) },
-	colony_status { colony_status } {}
+	colony_status { colony_status }, pop_type_distribution { &pop_type_keys } {}
+
+std::string State::get_identifier() const {
+	return StringUtils::append_string_views(state_set.get_region().get_identifier(), "_", owner->get_identifier());
```

Comment:
Note you can have a colonial state and non-colonial state with the same owner in the same area.
So maybe append `_c` or `_colony` if it is colonial.
When the colonial part becomes a state, they merge.